### ISSUE TITLE
git.present state failed to create repository

### ISSUE THREAD

**Issue by marx-freedom** [saltstack/salt#36881]
### Description of Issue/Question
I want to create an empty git repository with file permissions limited to owner and group only.
I've got exception when running `salt ... state.highstate`

```
          ID: saltmaster1 repo prepared
    Function: git.present
        Name: /srv/git/saltmaster1.git
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/site-packages/salt/state.py", line 1733, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1652, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/site-packages/salt/states/git.py", line 1750, in present
                  user=user)
                File "/usr/lib/python2.7/site-packages/salt/modules/git.py", line 1766, in init
                  ignore_retcode=ignore_retcode)['stdout']
                File "/usr/lib/python2.7/site-packages/salt/modules/git.py", line 299, in _git_run
                  raise CommandExecutionError(msg)
              CommandExecutionError: Command 'git init --bare --shared=0False /srv/git/saltmaster1.git' failed: fatal: bad config value for 'arg'
     Started: 17:22:41.421428
    Duration: 98.743 ms
     Changes:
```

### Setup
Salt minion is running under root.

State contains following configuration:

```
saltmaster1 repo prepared:
    git.present:
        - name: /srv/git/saltmaster1.git
        - bare: True
        - user: git
        - shared: false
```

### Steps to Reproduce Issue
salt '*' state.highstate

### Versions Report
On minion

```
Salt Version:
           Salt: 2016.3.3

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Sep 15 2016, 22:37:39)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-327.36.1.el7.x86_64
         system: Linux
        version: CentOS Linux 7.2.1511 Core
```

```
[root@jenkins jenkins]# git --version
git version 1.8.3.1
```

On Master:

```
Salt Version:
           Salt: 2016.3.3

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: 0.6.4
      gitpython: 2.0.8
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Sep 15 2016, 22:37:39)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-327.36.1.el7.x86_64
         system: Linux
        version: CentOS Linux 7.2.1511 Core

```

**Comment by gtmanfred** [comment]
Thanks for submitting the PR to fix this!

**Comment by rallytime** [comment]
Closed via #36912

**Comment by marx-freedom** [comment]
Thanks!
